codelabs/codelab-02/codelab2README.md

```diff
@@ -0,0 +1,501 @@
+
+![S3 Splash Page](../../../media/codelabs/codelab-02/s3.png)
+
+#### AWS CLI
+
+##### IAM User
+
+To work with the CLI, you'll need to first create an IAM user via the AWS Console. Open the "IAM" service and navigate to "Users" (tab on left) > "Add user".
+
+1. Pick a name for your user, and check "Programmatic Access". You won't need to sign in to the AWS console with your user.
+
+![Create User](../../../media/codelabs/codelab-02/create-user.png)
+
+2. You need to specify what permissions your new user has. To do this, we will create an IAM group for admin users. Give this group a name and select the "AdministratorAccess" policy.
+
+![Admin Permissions](../../../media/codelabs/codelab-02/admin-group.png)
+
+3. Click through to the "Complete" page where AWS will provide you with the access key ID and secret access key. Keep the latter safe, perhaps saving them in a text file on your machine, because AWS will not show it to you again. These keys are used to sign programmatic requests that you make to AWS.
+
+##### Testing the CLI
+
+If you haven't already done so, set up your local development environment. Follow the instructions here: [Environment Setup](https://github.com/UMD-CS-STICs/389Lspring18/blob/master/env.md). Upon entering the final command, you will be prompted to give the keys you just obtained. Set us-east-1 and json as the default region and output respectively.
+
+$ aws configure 
+AWS Access Key ID [None]: AKIAIOSFODNN7EXAMPLE 
+AWS Secret Access Key [None]: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY 
+Default region name [None]: us-east-1 
+Default output format [None]: json
+
+Now test your credentials with the following command.
+
+$ aws its get-caller-identity
+
+Good to go. We'll use the CLI more below.
+
+##### Root User
+
+To clarify the difference between the user you have been using to log in to the AWS Console and the IAM user you just created:
+
+The credentials you've been using to log in with are the root user credentials. Normally, you avoid using the root account and instead federate access out to IAM users so that you can audit their usage. However, in this class, we will continue to use the root account for simplicity.
+
+Essentially, you have two users for your account: the root user, which you use to access the AWS console, and the IAM user, which you use to programmatically access AWS. You won't need to think about this now that your AWS environment is configured, but it is good to know.
+
+### S3 Basics
+
+There are two main ways that you will programmatically interact with AWS. The first is through the CLI and the second is through the boto3 SDK. The CLI is great for one-time changes to your AWS environment. However, if you want to create dynamic scripts, then you will want to use something else — in this class, that will be Python with Amazon's Python SDK, boto3.
```

Comment:
Add a note that double-checks that they have installed the AWS cli, else referring them back to the env setup page